packages/core/src/module/module.ts

```diff
@@ -0,0 +1,40 @@
+import { DocumentNode } from 'graphql';
+import { ReflectiveInjector, Injector, Provider } from '@graphql-modules/di';
+import { moduleFactory, ModuleFactory } from './factory';
+import { ID, Single } from '../shared/types';
+import { TypeDefs, Resolvers } from './types';
+import { ModuleMetadata } from './metadata';
+import { ResolveMiddlewareMap } from '../shared/middleware';
+
+export interface ModuleConfig {
+  id: ID;
+  dirname?: string;
+  typeDefs: TypeDefs;
+  resolvers?: Resolvers;
+  resolveMiddlewares?: ResolveMiddlewareMap;
+  providers?: Provider[] | (() => Provider[]);
+}
+
+export type ModuleContext<T = {}> = {
```

Comment:
Let's move this to a namespace and then re-export it:
```
namespace GraphQLModules {
  export interface Context { 
    injector: Injector
  }
} 
export type ModuleContext = GraphQLModules.Context;
```

This way people can use declaration merging to extend the `ModuleContext` type.